### PR TITLE
Add experimental co.rerank API

### PR DESCRIPTION
**Pull request by lfayoux**
This PR adds support for the experimental co.rerank API. Access to this API is limited to a small number of users, who will be using the Python SDK.

### CODE REVIEW

File: cohere/rerank.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from typing import List, Optional, Dict, NamedTuple, Any, Iterator
+
+from cohere.response import CohereObject
+
+RerankDocument = NamedTuple("Document", [("text", str)])
```
Comment (mkozakov) [inline cohere/rerank.py:5]: let's add docs here as well via 
```
RerankDocument.__docs__ = """..."""

File: cohere/rerank.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+RerankDocument.__doc__ = """
+Returned by co.rerank,
+dict which always contain text but can also contain aribitrary fields
```
Comment (mkozakov) [inline cohere/rerank.py:8]:
```suggestion
dict which always contains text but can also contain aribitrary fields
```

**Review by mkozakov — APPROVED** [review]
Lets make sure we have docstrings for any classes that would be exposed to the user

File: cohere/client.py [hunk 3]
```diff
@@ -318,6 +318,38 @@ def feedback(self, id: str, good_response: bool, desired_response: str = "", fee
         self.__request(cohere.FEEDBACK_URL, json_body)
         return Feedback(id=id, good_response=good_response, desired_response=desired_response, feedback=feedback)
 
+    def rerank(self,
+               query: str,
+               documents: Union[List[str], List[Dict[str, Any]]],
+               top_n: int = None) -> Reranking:
+        """Returns an ordered list of documents oridered by their relevance to the provided query
```
Comment (amorisot) [inline cohere/client.py:325]:
```suggestion
        """Returns an ordered list of documents ordered by their relevance to the provided query
```

File: cohere/client.py [hunk 4]
```diff
@@ -318,6 +318,38 @@ def feedback(self, id: str, good_response: bool, desired_response: str = "", fee
...
+
+        Args:
+            query (str): The search query
+            documents (list[str], list[dict]): The documents to rerank
+            top_n (int): (optional) The number of results to return, defaults to return all results
```
Comment (amorisot) [inline cohere/client.py:330]:
```suggestion
            top_n (int): (optional) The number of results to return, defaults to returning all results
```

File: cohere/rerank.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
+from typing import List, Optional, Dict, NamedTuple, Any, Iterator
+
+from cohere.response import CohereObject
+
+RerankDocument = NamedTuple("Document", [("text", str)])
+RerankDocument.__doc__ = """
+Returned by co.rerank,
+dict which always contains text but can also contain aribitrary fields
```
Comment (amorisot) [inline cohere/rerank.py:8]:
```suggestion
dict which always contains text but can also contain arbitrary fields
```

**Review by sebastian-hofstaetter — APPROVED** [review]
lgtm

**Pull request merged**